2021_RTCSA_dynfed/src/dynfed.rs

```diff
@@ -48,6 +122,108 @@ where
 
     (minimum_cores, execution_order)
 }
+
+#[allow(dead_code)]
+pub fn dynamic_federated<T>(
+    dag_set: &mut [Graph<NodeData, i32>],
+    processor: &mut T,
+    scheduler: &mut impl SchedulerBase<T>,
+) -> i32
+where
+    T: ProcessorBase + Clone,
+{
+    let mut current_time = 0;
+    let dag_set_length = dag_set.len();
+    let mut dyn_fed_handlers = vec![DynamicFederatedHandler::new(); dag_set_length];
```

Comment:
Hashmaps are easier to understand.

Reply:
It is deprecated to hashmap if there is no change in performance.
Because, in order to modify the contents of a HashMap, you must use the get_mut method. This is because the hashmap owns the data and provides only an immutable reference to it.

For example.

> dyn_fed_handlers[dag_id].set_minimum_cores_and_execution_order(dag, scheduler);
is

>     if let Some(handler) = dyn_fed_handlers.get_mut(&dag_id) {
        handler.set_minimum_cores_and_execution_order(dag, scheduler);
    } else {
        // Handle the error here. For example, you can simply ignore the missing key:
        eprintln!("Key {} not found in dyn_fed_handlers", dag_id);
    }

Reply:
OK. I have understood.